### PR TITLE
Canonicalize C++ code as per style guide

### PR DESCRIPTION
**Pull request by changlan**
- Fix warnings from cpplint that are not false positives. 
- Run clang-format

### CODE REVIEW

File: core/module.h [hunk 1]
```diff
@@ -291,7 +292,7 @@ inline void Module::RunChooseModule(gate_idx_t ogate_idx,
   }
 
   ctx.set_current_igate(ogate->igate_idx());
-  ((Module *)ogate->arg())->ProcessBatch(batch);
+  (reinterpret_cast<Module *>(ogate->arg()))->ProcessBatch(batch);
```
Comment (barath) [inline core/module.h]: static_cast instead?

File: core/module_test.cc [hunk 2]
```diff
@@ -155,7 +155,7 @@ TEST_F(ModuleTester, RunCommand) {
     response = m->RunCommand("foo", arg);
     EXPECT_EQ(0, response.error().err());
   }
-  EXPECT_EQ(10, ((AcmeModule *)m)->n);
+  EXPECT_EQ(10, (reinterpret_cast<AcmeModule *>(m))->n);
```
Comment (barath) [inline core/module_test.cc]: static_cast instead?

File: core/debug.cc [hunk 3]
```diff
@@ -6,6 +6,10 @@
 #include <ucontext.h>
 #include <unistd.h>
 
+#include <glog/logging.h>
```
Comment (changlan) [inline core/debug.cc]: Looks like it isn't. Functions in debug.h still get called.

**Comment by barath** [comment]
image

**Pull request merged**